docs/advanced/additional_requests.rst

```diff
@@ -0,0 +1,270 @@
+.. _`advanced-requests`:
+
+===================
+Additional Requests
+===================
+
+Websites nowadays needs a lot of page interactions to display or load some key
+information. In most cases, these are done via AJAX requests. Some examples of these are:
+
+    * Clicking a button on a page to reveal other similar products.
+    * Clicking the `"Load More"` button to retrieve more images of a given item.
+    * Scrolling to the bottom of the page to load more items `(i.e. infinite scrolling)`.
+    * Hovering that reveals a tool-tip containing additional page info.
+
+As such, performing additional requests inside Page Objects are inevitable to
+properly extract data for some websites.
+
+.. warning::
+
+    Additional requests made inside a Page Object aren't meant to represent
+    the **Crawling Logic** at all. They are simply a low-level way to interact
+    with today's websites which relies on a lot of page interactions to display
+    its contents.
+
+
+HttpClient
+==========
+
+The main interface for executing additional requests would be :class:`~.HttpClient`.
+It also has full support for :mod:`asyncio` enabling developers to perform
```

Comment:
Does it (it - page object's to_item method?) support asyncio or coroutines? For example, if it supports asyncio, I'd expect asyncio libraries to work, e.g. asyncio.gather, or asyncio.wait.

Reply:
Yes. Added more explicit notes on this in https://github.com/scrapinghub/web-poet/pull/22/commits/f0a3419325508c4e8e4b5bfec87a1d19aaacc660#diff-1b9a5e2fee681a2590fc89977cef10683e746d3fe4b2d207c262a75e17ec9657R119-R122